packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -307,8 +308,13 @@ class AndroidStudio implements Comparable<AndroidStudio> {
     if (!globals.processManager.canRun(javaExecutable)) {
       _validationMessages.add('Unable to find bundled Java version.');
     } else {
-      final RunResult result = processUtils.runSync(<String>[javaExecutable, '-version']);
-      if (result.exitCode == 0) {
+      RunResult result;
+      try {
+        result = processUtils.runSync(<String>[javaExecutable, '-version']);
```

Comment:
While you're here, would it be possible to switch this to `run` instead of `runSync`? I know we have https://github.com/flutter/flutter/issues/7102 sitting around, though I'm not sure how relevant it is

Reply:
Hmm good point. Unfortunately `_init()` is called from various constructors, but I'll see if I can rearrange a bit.

Reply:
Okay. I've been pulling on this thread for a couple of hours. I remember now why I gave up the last time I tried this. The assumption that the Android SDK is available synchronously is pervasive throughout the tool. One place the tool looks for the SDK is from Android Studio, so if Android Studio lookup is asynchronous then everything related to the Android SDK has to be made asynchronous as well. It's probably a good 2-3 days of solid work to untangle. I think I should land this PR as is, and continue to ponder the bigger refactor.